tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/TeamMailboxRevokeAccessMethodContract.scala

```diff
@@ -0,0 +1,216 @@
+      .body(
+        s"""{
+           |	"using": ["urn:ietf:params:jmap:core", "urn:ietf:params:jmap:mail", "com:linagora:params:jmap:team:mailboxes"],
+           |	"methodCalls": [
+           |		["TeamMailbox/revokeAccess", {
+           |			"accountId": "$BOB_ACCOUNT_ID",
+           |			"ids": ["nonExistTeamMailbox@domain.tld"]
+           |		}, "c0"]
+           |	]
+           |}""".stripMargin)
+    .when
+      .post
+    .`then`
+      .log().ifValidationFails()
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract
+      .body
+      .asString
+
+    assertThatJson(response).isEqualTo(
+      s"""{
+         |    "sessionState": "${SESSION_STATE.value}",
+         |    "methodResponses": [
+         |        [
+         |            "TeamMailbox/revokeAccess",
+         |            {
+         |                "accountId": "$BOB_ACCOUNT_ID",
+         |                "notRevoked": {
+         |                    "nonExistTeamMailbox@domain.tld": {
+         |                        "type": "notFound",
+         |                        "description": "#TeamMailbox:team-mailbox@domain.tld:nonExistTeamMailbox can not be found"
+         |                    }
+         |                }
+         |            },
+         |            "c0"
+         |        ]
+         |    ]
+         |}""".stripMargin)
+  }
+
+  @Test
+  def givenBobDoesNotHaveAccessToTeamMailboxThenRevokeAccessShouldSucceed(server: GuiceJamesServer): Unit = {
+    val teamMailbox = TeamMailbox(DOMAIN, TeamMailboxName("hiring"))
+    server.getProbe(classOf[TeamMailboxProbe])
+      .create(teamMailbox)
+
+    val response = `given`
+      .body(
+        s"""{
+           |	"using": ["urn:ietf:params:jmap:core", "urn:ietf:params:jmap:mail", "com:linagora:params:jmap:team:mailboxes"],
+           |	"methodCalls": [
+           |		["TeamMailbox/revokeAccess", {
+           |			"accountId": "$BOB_ACCOUNT_ID",
+           |			"ids": ["hiring@domain.tld"]
+           |		}, "c0"]
+           |	]
+           |}""".stripMargin)
+    .when
+      .post
+    .`then`
+      .log().ifValidationFails()
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract
+      .body
+      .asString
+
+    assertThatJson(response).isEqualTo(
+      s"""{
+         |	"sessionState": "${SESSION_STATE.value}",
+         |	"methodResponses": [
+         |		[
+         |			"TeamMailbox/revokeAccess",
+         |			{
+         |				"accountId": "$BOB_ACCOUNT_ID",
+         |				"revoked": ["hiring@domain.tld"]
+         |			},
+         |			"c0"
+         |		]
+         |	]
+         |}""".stripMargin)
+
+    assertThat(server.getProbe(classOf[TeamMailboxProbe]).listMembers(teamMailbox).asJava)
+      .isEmpty()
+  }
+
+  // TODO invalid mailbox name case should be fail
+  // TODO user revoke the same team mailbox 2 times (should it be idempotent?)
+  // TODO mixed case succeed and fail case
+  // TODO make sure delegatee can not access this API
```

Comment:
Also add a test to unsure one no longer see in mailboxes/get a revoked team mailbox...